src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3261,7 +3272,18 @@ private void VisitObjectElementInitializer(int containingSlot, BoundAssignmentOp
         private new void VisitCollectionElementInitializer(BoundCollectionElementInitializer node)
         {
             // Note: we analyze even omitted calls
-            (var reinferredMethod, _, _) = VisitArguments(node, node.Arguments, refKindsOpt: default, node.AddMethod, node.ArgsToParamsOpt, node.DefaultArguments, node.Expanded, node.InvokedAsExtensionMethod);
+            (var reinferredMethod, _, _) = VisitArguments(
+                node,
+                node.Arguments,
+                refKindsOpt: default,
+                node.AddMethod,
+                node.ArgsToParamsOpt,
+                node.DefaultArguments,
+                node.Expanded,
+                node.ImplicitReceiverOpt,
+                receiverState: NullableFlowState.NotNull,
```

Comment:
It feels like whenever we might have a not-null receiver, we pass a NotNull flow state here. Is it necessary for this to be its own parameter?